doc/decisions/iterating_name_parts.md

```diff
@@ -0,0 +1,57 @@
+# Iterating Keyname Parts
+
+## Problem
+
+While iterating over the parts of a keyname is easy (just jump from `\0` to `\0` until you hit the name size), there is no good API for it.
+
+## Constraints
+
+No changes to the underlying APIs and
+
+## Assumptions
+
+## Considered Alternatives
+
+- Put the new function into `libelektra-core`.
+
+## Decision
+
+Add this new function to `libelektra-operations` (see [Operations Library](operations_library.md)):
+
+```c
+/**
+ * Iterate over the key name parts of @p key.
+ *
+ * @pre @p currentPart, must be `NULL` or point anywhere into the unescaped name of @p key
+ *
+ * This function defins the "next part" after @p currentPart as:
+ * - the first part of @p key (after namespace), if `currentPart == NULL`
+ * - `NULL`, if @p currentPart is the last part of @p key, i.e. `currentPart == keyBaseName (key)`
+ * - the next part within @p key after the location @p currentPart points to
+ *
+ * To iterate over the parts of a key you can use:
+ * @code{.c}
+ * Key * k;
+ * for (const char * part = keyGetNextPart (k, NULL); part != NULL; part = keyGetNextPart (k, part))
```

Comment:
> `keyGetNextPart` for escaped key names would be more interesting, wouldn't it?

You may want the unescaped part for various reasons and implementing it this is a lot easier. The user can always call the escaping function on the returned string. Since the escaped name won't be stored anymore, if we returned the escaped name we would basically do the same anyway.

I actually think, the unescaped name is a lot more useful. It is the actual value of the name part, not just a representation.

> Shouldn't we put all the key name escaping stuff in a separate library? (operations already looks oversized)

I'm not sure what you are referring to here, but all the (un)escaping code has to part of `libelektra-core`. Unless of course the core now only supports unescaped names. That would be pretty big change, but then moving the (un)escaping to a separate library would make sense.

Reply:
> I'm not sure what you are referring to here, but all the (un)escaping code has to part of libelektra-core. Unless of course the core now only supports unescaped names. That would be pretty big change, but then moving the (un)escaping to a separate library would make sense.

Exactly, I am referring to to-be-clarified decisions about if and how (un)escaped of key names should be in core or in an extra library. As these parts also need bindings, it could be libelektra-operations (in the proposed terminology).

The big question is the overall dependencies: at least for some parts of Elektra the core alone must suffice, otherwise the core lib would be useless.